ckanext/datastore/blueprint.py

```diff
@@ -92,25 +101,17 @@ def dump(resource_id: str):
     content_type = None
     content_disposition = None
 
-    if fmt == 'csv':
-        content_disposition = 'attachment; filename="{name}.csv"'.format(
-                                    name=resource_id)
-        content_type = b'text/csv; charset=utf-8'
-    elif fmt == 'tsv':
-        content_disposition = 'attachment; filename="{name}.tsv"'.format(
-                                    name=resource_id)
-        content_type = b'text/tab-separated-values; charset=utf-8'
-    elif fmt == 'json':
-        content_disposition = 'attachment; filename="{name}.json"'.format(
-                                    name=resource_id)
-        content_type = b'application/json; charset=utf-8'
-    elif fmt == 'xml':
-        content_disposition = 'attachment; filename="{name}.xml"'.format(
-                                    name=resource_id)
-        content_type = b'text/xml; charset=utf-8'
+    for d in dump_formats():
+        if fmt == d.get_format():
+            format_class = d
+            break
```

Comment:
a `{format_name: format_class}` dict would be better for the formats so we can look them up or get all the format names instantly.